### References
1.	[Saini, S., Singh, J. and Saini, M., 2014. An Improved Branch and Bound Algorithm for Cyclopeptide Sequencing. Biom. Bioinform., 6(6), pp.154-158.](https://scholar.google.com/citations?view_op=view_citation&hl=en&user=pQywCDkAAAAJ&citation_for_view=pQywCDkAAAAJ:u-x6o8ySG0sC)

2.	[Klepeis, J.L., Floudas, C.A., Morikis, D. and Lambris, J.D., 1999. Predicting peptide structures using NMR data and deterministic global optimization. Journal of Computational Chemistry, 20(13), pp.1354-1370.](https://onlinelibrary.wiley.com/doi/abs/10.1002/%28SICI%291096-987X%28199910%2920%3A13%3C1354%3A%3AAID-JCC3%3E3.0.CO%3B2-N)

3.	[Tan, N.H. and Zhou, J., 2006. Plant cyclopeptides. Chemical reviews, 106(3), pp.840-895.](https://pubs.acs.org/doi/abs/10.1021/cr040699h)

4.	[Mohimani, H., 2013. Bioinformatics Methods for Natural Product Discovery. University of California, San Diego.](https://escholarship.org/uc/item/1cp8f9f1)

5.	[Palagi, P.M., Arhné, E., Müller, M. and Lisacek, F., 2011. Bioinformatics tools for detecting post-translational modifications in mass spectrometry data. Amino Acids, 5(14), pp.463-475](https://books.google.co.in/books?hl=en&lr=&id=m5gIgKeuUmAC&oi=fnd&pg=PA463&dq=Palagi,+P.M#v=onepage&q=Palagi%2C%20P.M&f=false)
